File: LeetCode/LeetCode 75/392_Is_Subsequence/392_Is_Subsequence.py
from typing import *

class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if len(s) > len(t):
            return False
        if len(s) == 0:
            return True
        
        i = 0
        for x in t:
            if s[i] == x:
                i += 1
            if i == len(s):
                return True
        return True if i == len(s) else False
        

s = Solution()

print(s.isSubsequence("abc", "ahdgbc"))
print(s.isSubsequence("axc", "ahdgbc"))
print(s.isSubsequence("", "ahdgbc"))

File: LeetCode/LeetCode 75/392_Is_Subsequence/392_Is_Subsequence.kt
class Solution {
    fun isSubsequence(s: String, t: String): Boolean {
        if (s.length == 0) return true
        if (s.length > t.length) return false

        var i = 0
        for ((j, _) in t.withIndex()) {
            if (i != s.length && s[i] == t[j]) i += 1
        }
        return if (i == s.length) true else false
    }
}

fun main() {
    val s = Solution()

    println(s.isSubsequence("abc", "ahdgbc"))
    println(s.isSubsequence("axc", "ahdgbc"))
    println(s.isSubsequence("", "abdcef"))
}